_posts/2019-08-13-kubernetes-gated-deployments.md

```diff
@@ -0,0 +1,280 @@
+      secretName: newrelic-secrets
+      secretKey: example-rest-service
+      appName: example-rest-service
+      minSamples: 50
+      maxTime: 600
+      testPath: /shopper/products
+```
+
+This plugin retrieves response times from New Relic for the control and
+treatment versions, and tests if they are statistically significantly different
+using the [Mann-Whitney U
+test](https://en.wikipedia.org/wiki/Mann–Whitney_U_test). Because statistical
+significance does not necessarily imply practical significance, the plugin
+exposes a threshold setting before it marks the treatment version as causing
+harm to the metric of response time. This is useful in the case when the
+application does not receive much traffic, as small sample sizes can result in
+erroneous statistically significant results. Depending on the traffic patterns
+and response time distribution of the application, teams can also configure a
+minimum number of samples before analyzing the response times, and a maximum
+amount of time in seconds that the experiment will run for before automatically
+deciding that it is a success.
+
+## Kubernetes Gated Deployments in production
+
+We have replaced log analysis of a single canary pod that some of our services
+used with Kubernetes Gated Deployments, and we have been using the New Relic
+response time decision plugin for a while with some production services. It has
+already detected and rolled back a performance regression that resulted from
+calling a new API with increased response times and has given engineers more
+confidence in their deployed changes.
+
+## Learn more and get started
+
+With Kubernetes Gated Deployments, engineers are able to perform automated
+regression testing and canary analysis with their continuous deployments, and
+limit any negative impact resulting from their changes. By configuring this
+Kubernetes extension with the functionality and performance metrics associated
+with their application, they can rely on the controller to measure and analyze
+those metrics in a statistically sound way, and automatically mitigate any
+negative impact. Kubernetes Gated Deployments does not require any application
+code changes, making it simple to apply to existing deployment pipelines.
+
+If you want to learn more about or get started with Kubernetes Gated
+Deployments, please visit
+<https://github.com/godaddy/kubernetes-gated-deployments>. We welcome
+contributions and would love to expand the decision plugins available to support
+other metric backends and analysis techniques.
+
+Our team of engineers and full-stack data scientists builds experimentally
+validated services. Come join us and contribute to services that help us
+leverage experimentation services and machine learning models to improve the
+GoDaddy customer experience. We're looking for a
+[principal](https://godaddy.wd1.myworkdayjobs.com/GoDaddy_careers/job/Kirkland/Principal-Software-Engineering----Machine-Learning-and-Experimentation_R006162-1)
```

Comment:
@stevenkfu Sorry, I gave you the wrong links. Can you swap in these instead please?

https://careers.godaddy.com/job/cambridge/principal-software-development-engineer-machine-learning-and-experimentation/18045/11812427

https://careers.godaddy.com/job/cambridge/senior-software-engineering-machine-learning-and-experimentation/18045/12444210 and

Reply:
updated